### DISCUSSION TITLE
How to get boundary of a thbSpline?

### DISCUSSION THREAD

**Discussion by Rennngw** [gismo/gismo#584] [Q&A] [answered]
Hello, i'm trying to get boundary of a thbSpline. But I found that the result I got was empty, why did I get such a result?
My sample code is as follows.
Thanks！

**Get geometry:**
```
std::string geo_path = "plateWithHole.xml";
gsGeometry<>::uPtr mpPtr = gsReadFile<>(geo_path);
gsTensorBSplineBasis<2, real_t>::uPtr mbaseptr = gsReadFile<>(geo_path);

auto& patch = *mpPtr;
auto& tensorbas = *mbaseptr;

gsTensorBSpline<2, double> bSpline(tensorbas,patch.coefs());
bSpline.degreeElevate();
for (index_t i = 0; i < 2; ++i)
   bSpline.uniformRefine();

gsTHBSpline<2, double> thbSpline(bSpline);
```

**Get boundary:**
```
gsMatrix<index_t> ind = thbSpline.basis().boundary(boundary::side::north);
auto northCrv = thbSpline.boundary(boundary::side::north);
```
**Result:**
"ind" is empty 
 `auto northCrv = thbSpline.boundary(boundary::side::north);` this code will throw an error.

If i running code like this:
```
std::string geo_path = "plateWithHole.xml";
gsGeometry<>::uPtr mpPtr = gsReadFile<>(geo_path);
gsTensorBSplineBasis<2, real_t>::uPtr mbaseptr = gsReadFile<>(geo_path);
auto& patch = *mpPtr;
auto& tensorbas = *mbaseptr;
gsTensorBSpline<2, double> bSpline(tensorbas,patch.coefs());
bSpline.degreeElevate();
for (index_t i = 0; i < 2; ++i)
   bSpline.uniformRefine();
gsMatrix<index_t> ind = bSpline.basis().boundary(boundary::side::north);
auto northCrv = bSpline.boundary(boundary::side::north);
```
"ind" is not empty, and i can get boundary of bSpline.

So how can i get boundary of thbSpline?
Thanks!
[plateWithHole.zip](https://github.com/gismo/gismo/files/10705130/plateWithHole.zip)

**Comment by ckarampa** [comment, marked as answer]
As with your other question, I cannot replicate it, the code runs fine.

**Reply by Rennngw** [reply]
Thank you for your reply.
I think this problem comes from the [problem](https://github.com/gismo/gismo/discussions/586#discussion-4845094) when I use bspline to initialize thbspline, so if thbspline can be constructed normally, this problem will also be solved.
Thank you again for your help.

Best regards
Rennngw
